### PR TITLE
File Append E2E test pass

### LINKED ISSUE #8019: FileAppendSuite

### PR DESCRIPTION
**Pull request by povolev15**
I added Fees calculation to file append in order to E2E to pass.

Fixes #8019

### CODE REVIEW

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeCalculatorImpl.java [hunk 1]
```diff
@@ -135,6 +135,13 @@ public FeeCalculator addNetworkRamByteSeconds(long amount) {
         return this;
     }
 
+    @NonNull
+    @Override
+    public FeeCalculator addStorageBytesHours(long seconds) {
```
Comment (jsync-swirlds) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeCalculatorImpl.java:140]: If this is always seconds, and the underlying call is seconds, why is it named `addStorageBytesHours`?

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/NumbersModule.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package com.hedera.node.app.info;
+
+import com.hedera.node.app.spi.numbers.HederaFileNumbers;
+import dagger.Binds;
+import dagger.Module;
+
+@Module
+public interface NumbersModule {
```
Comment (jsync-swirlds) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/NumbersModule.java:24]: For clarity, could we name this `FileNumbersModule`, so it's clear we're supporting file numbers?

File: hedera-node/hedera-config/src/main/java/com/hedera/node/config/converter/LongPairConverter.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ */
+public class LongPairConverter implements ConfigConverter<LongPair> {
+
+    private static final String PATTERN = Pattern.quote("-");
+
+    @Override
+    public LongPair convert(final String value) throws IllegalArgumentException, NullPointerException {
+        Objects.requireNonNull(value, "Parameter 'value' cannot be null");
+        final String[] split = value.split(PATTERN, 2);
+        if (split.length == 1) {
```
Comment (jsync-swirlds) [inline hedera-node/hedera-config/src/main/java/com/hedera/node/config/converter/LongPairConverter.java:35]: This probably should be `<=`, in case there is no matching separator.

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java [hunk 4]
```diff
@@ -140,6 +148,29 @@ public void handle(@NonNull final HandleContext handleContext) throws HandleExce
...
+                (fileAppend.contents() != null) ? fileAppend.contents().length() : 0;
+
+        final long effectiveLifeTime;
+        if (fileNumbers.isSoftwareUpdateFile(file.fileId().fileNum())) {
+            effectiveLifeTime = THREE_MONTHS_IN_SECONDS;
+        } else {
+            final var effCreationTime =
+                    handleContext.body().transactionID().transactionValidStart().seconds();
+            final var effExpiration = (file.expirationSecond() > 0) ? file.expirationSecond() : effCreationTime;
+            effectiveLifeTime = effExpiration - effCreationTime;
```
Comment (jsync-swirlds) [inline hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java:162]: This produces an effective lifetime of `0` when expiration is not set.  Is that the current behavior?  It seems odd that the default lifetime for fee calculation would be 0.

File: hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileAppendTest.java [hunk 5]
```diff
@@ -80,16 +87,31 @@ class FileAppendTest extends FileTestBase {
...
-        lenient().when(preHandleContext.configuration()).thenReturn(testConfig);
-        lenient().when(handleContext.configuration()).thenReturn(testConfig);
+        when(preHandleContext.configuration()).thenReturn(testConfig);
+        when(handleContext.configuration()).thenReturn(testConfig);
+        when(handleContext.feeCalculator(any(SubType.class))).thenReturn(feeCalculator);
+        when(feeCalculator.calculate()).thenReturn(Fees.FREE);
+        when(feeCalculator.addBytesPerTransaction(anyLong())).thenReturn(feeCalculator);
+        when(feeCalculator.addStorageBytesHours(anyLong())).thenReturn(feeCalculator);
+        when(handleContext.feeAccumulator()).thenReturn(feeAccumulator);
+        when(feeCalculator.legacyCalculate(any())).thenReturn(Fees.FREE);
```
Comment (jsync-swirlds) [inline hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileAppendTest.java:114]: Instead of mocking these, could we have used the FakeFeeCalculator and FakeFeeAccumulator from app-spi test fixtures?

File: hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileCreateTest.java [hunk 6]
```diff
@@ -122,7 +123,8 @@ private TransactionBody newCreateTxn(KeyList keys, long expirationTime) {
     void setUp() {
         subject = new FileCreateHandler();
         fileStore = new WritableFileStore(writableStates);
-        config = new FilesConfig(101L, 121L, 112L, 111L, 122L, 102L, 123L, 1000000L, 1024, 150L);
+        config = new FilesConfig(
+                101L, 121L, 112L, 111L, 122L, 102L, new LongPair(150L, 159L), 123L, 1000000L, 1024, 150L);
```
Comment (jsync-swirlds) [inline hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileCreateTest.java:127]: I believe you could be using HederaTestConfigBuilder to avoid having to fill all these in...

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeCalculatorImpl.java (hunk 1 above)
Comment (povolev15) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/FeeCalculatorImpl.java:140]: I think it comes from some abbreviation used in the mono code. I can change it to secs.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/NumbersModule.java (hunk 2 above)
Comment (povolev15) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/NumbersModule.java:24]: done thanks

File: hedera-node/hedera-config/src/main/java/com/hedera/node/config/converter/LongPairConverter.java (hunk 3 above)
Comment (povolev15) [reply hedera-node/hedera-config/src/main/java/com/hedera/node/config/converter/LongPairConverter.java:35]: done thanks

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java (hunk 4 above)
Comment (povolev15) [reply hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java:162]: @Neeharika-Sompalli  I think that what we saw with you.
That what I also see in methods metaForFileAppend

File: hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileAppendTest.java (hunk 5 above)
Comment (povolev15) [reply hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileAppendTest.java:114]: I tried originally I think, and it didn't work for me. I don't remember why I can try again.
I see we use the same in Consensus

Comment (jsync-swirlds) [reply hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileAppendTest.java:114]: The fake versions might not have all the methods implemented; so the objects might need to be updated, but I suspect that making those work for the general case might be more helpful than mocking in each service.

File: hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileCreateTest.java (hunk 6 above)
Comment (povolev15) [reply hedera-node/hedera-file-service-impl/src/test/java/com/hedera/node/app/service/file/impl/test/handlers/FileCreateTest.java]: needed in order not to use the default values

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java (hunk 4 above)
Comment (jsync-swirlds) [reply hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java:162]: Looks like we need to leave as-is for now, but please ensure we revisit this item in the future (after the modularization parity testing finishes).

**Review by jsync-swirlds — DISMISSED** [review]

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java (hunk 4 above)
Comment (povolev15) [reply hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java:166]: I added a comment for it

**Review by jsync-swirlds — DISMISSED** [review]

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/HederaFileNumbersImpl.java [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+import com.hedera.node.app.spi.numbers.HederaFileNumbers;
+import com.hedera.node.config.ConfigProvider;
+import com.hedera.node.config.data.FilesConfig;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+public class HederaFileNumbersImpl implements HederaFileNumbers {
```
Comment (netopyr) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/HederaFileNumbersImpl.java:27]: Why was this class added? It just returns what's already available in `FilesConfig`, right?

**Review by iwsimon — DISMISSED** [review]

**Review by netopyr — CHANGES_REQUESTED** [review]
I think `HederaFileNumbers` is not needed. The `FileAppendHandler` can get the configuration from context and work with the `FileConfig` directly.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/HederaFileNumbersImpl.java (hunk 7 above)
Comment (povolev15) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/HederaFileNumbersImpl.java:27]: you are right I fixed it and use the config directly.

Comment (povolev15) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/HederaFileNumbersImpl.java]: fixed and committed

**Review by jsync-swirlds — APPROVED** [review]

**Review by iwsimon — APPROVED** [review]

File: hedera-node/hedera-config/src/test/java/com/hedera/node/config/converter/LongPairConverterTest.java [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        // when
+        final var result = converter.convert(value);
+
+        // then
+        assertThat(result.left()).isEqualTo(1L);
+        assertThat(result.right()).isEqualTo(2L);
+    }
+
+    @Test
+    void testInvalidValueMultipleSemicolon() {
```
Comment (netopyr) [inline hedera-node/hedera-config/src/test/java/com/hedera/node/config/converter/LongPairConverterTest.java:56]: The name of this test is slightly misleading. 🙂

File: hedera-node/hedera-config/src/test/java/com/hedera/node/config/converter/LongPairConverterTest.java [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+        final Configuration configuration = ConfigurationBuilder.create()
+                .withConverter(new LongPairConverter())
+                .withSource(new SimpleConfigSource("testProperty", "1-2"))
+                .build();
+
+        // when
+        final List<LongPair> values = configuration.getValues("testProperty", LongPair.class);
+
+        // then
+        assertThat(values).isNotNull().hasSize(1);
```
Comment (netopyr) [inline hedera-node/hedera-config/src/test/java/com/hedera/node/config/converter/LongPairConverterTest.java:104]: I think the `isNotNull()` check is not needed because this is automatically done by `hasSize()` (and similar methods).

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java [hunk 10]
```diff
@@ -140,6 +143,42 @@ public void handle(@NonNull final HandleContext handleContext) throws HandleExce
...
+        }
+
+        final var fees = handleContext
+                .feeCalculator(SubType.DEFAULT)
+                .addBytesPerTransaction(BASIC_ENTITY_ID_SIZE + dataLength)
+                .addStorageBytesSeconds(dataLength * effectiveLifeTime)
+                .calculate();
+
+        handleContext.feeAccumulator().charge(handleContext.payer(), fees);
```
Comment (netopyr) [inline hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java:179]: This has to be done differently now. The initial example was wrong. Just a heads up, should probably be done in a separate PR.

**Review by netopyr — APPROVED** [review]
LGTM

File: hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java (hunk 10 above)
Comment (povolev15) [reply hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileAppendHandler.java:179]: can you attach a right example and I will open a ticket for it.

File: hedera-node/hedera-config/src/test/java/com/hedera/node/config/converter/LongPairConverterTest.java (hunk 8 above)
Comment (povolev15) [reply hedera-node/hedera-config/src/test/java/com/hedera/node/config/converter/LongPairConverterTest.java:56]: fixed thanks

File: hedera-node/hedera-config/src/test/java/com/hedera/node/config/converter/LongPairConverterTest.java (hunk 9 above)
Comment (povolev15) [reply hedera-node/hedera-config/src/test/java/com/hedera/node/config/converter/LongPairConverterTest.java:104]: fixed thanks

**Review by jsync-swirlds — APPROVED** [review]

**Review by iwsimon — APPROVED** [review]

**Review by netopyr — APPROVED** [review]
LGTM
